### PR TITLE
Add some chest effects and change textures

### PR DESCRIPTION
**Pull request by LoneWolfHT**
## **A list of things in this PR**
* A spooky_effects mod to hold functions that do things like spawn ghost sprites.
* A chest_changes mod.
  * Sets the textures of the treasure chests to the textures in MTG's bone mod
  * Treasure chests, when opened, 'release' a ghost that floats into the air and disappears
  * Treasure chests, when emptied, 'release' an angry ghost that dashes towards the players face

### CODE REVIEW

File: chest_changes/init.lua [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+		minetest.log("action", player:get_player_name() ..
+			" takes stuff from chest at " .. minetest.pos_to_string(pos))
+		local inv = minetest.get_inventory({type = "node", pos=pos})
+		if not inv or inv:is_empty("main") then
+			minetest.set_node(pos, {name="air"})
+			minetest.show_formspec(player:get_player_name(), "", player:get_inventory_formspec())
+
+			spooky_effects.spawn_angry_ghost(pos, player)
+		end
+	end,
```
Comment (LoneWolfHT) [inline chest_changes/init.lua:15]: I'll do my best to keep this compatible with the tsm_chests mod. Although I don't think there will be very many changes

**Review by LoneWolfHT — COMMENTED** [review]
I can't find any problems

**Comment by LoneWolfHT** [comment]
DeltaForce is going to create a PR to this PR. Don't merge yet

**Comment by LoneWolfHT** [comment]
Actualy nvm. He could just create a PR to the main repo :thinking:

File: chest_changes/init.lua [hunk 2]
```diff
@@ -0,0 +1,16 @@
+minetest.override_item("tsm_chests:chest", {
+	on_rightclick = function(pos)
+		spooky_effects.spawn_ghost(pos)
+	end,
```
Comment (ClobberXD) [inline chest_changes/init.lua:4]: Have you tested whether overriding `on_rightclick` prevent the node formspec from being shown?

Comment (LoneWolfHT) [reply chest_changes/init.lua:4]: It still shows

Comment (ClobberXD) [reply chest_changes/init.lua:4]: Ok, thanks.

**Pull request merged**